app/controllers/service_controller.rb

```diff
@@ -19,10 +19,14 @@ class ServiceController < ApplicationController
 
   def button
     if params[:pressed] == "custom_button"
-      ids = if @display == 'generic_objects'
-              @lastaction == 'generic_object' ? @sb[:rec_id] : 'LIST'
-            end
-      custom_buttons(ids)
+      display_options = {}
```

Comment:
`please, try to keep the button method basically a `case` statement

if you need generic objects specific behavior here, create d merhod for that.

it's more readable this way and more testable too

thx!